### PR TITLE
fix tool/update_clang_options.zig for ArrayHashMap's Separated Key/Va…

### PR DESCRIPTION
**Pull request by redj**
…lue Storage breaking change.

### CODE REVIEW

**Comment by redj** [comment]
Is this the proper fix? I did run the tool and it generated an identical src/clang_options_data.zig file.

**Review by kubkon — APPROVED** [review]
LGTM!

File: tools/update_clang_options.zig [hunk 1]
```diff
@@ -412,16 +412,16 @@ pub fn main() anyerror!void {
     {
         var it = root_map.iterator();
         it_map: while (it.next()) |kv| {
-            if (kv.key.len == 0) continue;
-            if (kv.key.*[0] == '!') continue;
-            if (kv.value.* != .Object) continue;
-            if (!kv.value.Object.contains("NumArgs")) continue;
-            if (!kv.value.Object.contains("Name")) continue;
+            if (kv.key_ptr.*.len == 0) continue;
```
Comment (kubkon) [inline tools/update_clang_options.zig:415]:
```suggestion
            if (kv.key_ptr.len == 0) continue;
```

File: tools/update_clang_options.zig [hunk 2]
```diff
@@ -412,16 +412,16 @@ pub fn main() anyerror!void {
...
+            if (kv.key_ptr.*[0] == '!') continue;
+            if (kv.value_ptr.* != .Object) continue;
+            if (!kv.value_ptr.*.Object.contains("NumArgs")) continue;
```
Comment (kubkon) [inline tools/update_clang_options.zig:418]:
```suggestion
            if (!kv.value_ptr.Object.contains("NumArgs")) continue;
```

File: tools/update_clang_options.zig [hunk 3]
```diff
@@ -412,16 +412,16 @@ pub fn main() anyerror!void {
...
+            if (!kv.value_ptr.*.Object.contains("Name")) continue;
```
Comment (kubkon) [inline tools/update_clang_options.zig:419]:
```suggestion
            if (!kv.value_ptr.Object.contains("Name")) continue;
```

File: tools/update_clang_options.zig [hunk 4]
```diff
@@ -412,16 +412,16 @@ pub fn main() anyerror!void {
...
             for (blacklisted_options) |blacklisted_key| {
-                if (std.mem.eql(u8, blacklisted_key, kv.key.*)) continue :it_map;
+                if (std.mem.eql(u8, blacklisted_key, kv.key_ptr.*)) continue :it_map;
             }
-            if (kv.value.Object.get("Name").?.String.len == 0) continue;
-            try all_objects.append(&kv.value.Object);
+            if (kv.value_ptr.*.Object.get("Name").?.String.len == 0) continue;
```
Comment (kubkon) [inline tools/update_clang_options.zig:423]:
```suggestion
            if (kv.value_ptr.Object.get("Name").?.String.len == 0) continue;
```

File: tools/update_clang_options.zig [hunk 5]
```diff
@@ -412,16 +412,16 @@ pub fn main() anyerror!void {
...
+            try all_objects.append(&kv.value_ptr.*.Object);
```
Comment (kubkon) [inline tools/update_clang_options.zig:424]:
```suggestion
            try all_objects.append(&kv.value_ptr.Object);
```

**Review by kubkon — CHANGES_REQUESTED** [review]
Actually, on second thought, explicit dereferencing when accessing members or union variants are not needed - the pointer will get dereferenced automatically.

**Comment by redj** [comment]
@kubkon, thanks for the review. I think I got them all. I amended and force pushed. should be good to merge.

**Review by kubkon — APPROVED** [review]

**Pull request merged**